main.go

```diff
@@ -555,82 +572,168 @@ func HotelSearch(w http.ResponseWriter, req *http.Request) {
 	}
 
-	respData.Data = []jsonHotel{}
+	hotelCols := []string{
+		"country",
+		"city",
+		"state",
+		"address",
+		"name",
+		"description",
+		"type",
+	}
+	var respData jsonHotelSearchResp
 	for results.Next() {
-		res, _ := globalCollection.LookupIn(results.Row().ID, []gocb.LookupInSpec{
-			gocb.GetSpec("country", nil),
-			gocb.GetSpec("city", nil),
-			gocb.GetSpec("state", nil),
-			gocb.GetSpec("address", nil),
-			gocb.GetSpec("name", nil),
-			gocb.GetSpec("description", nil),
-		}, nil)
-		// We ignore errors here since some hotels are missing various
-		//  pieces of data, but every key exists since it came from FTS.
+		res, err := hotels.Get(results.Row().ID, &gocb.GetOptions{
+			Project: hotelCols,
+		})
+		if err != nil {
+			app.logger.Printf("Error occurred fetching hotel: %s", err)
+			writeJsonFailure(w, 500, err)
+			return
+		}
 
 		var hotel jsonHotel
-		res.ContentAt(0, &hotel.Country)
-		res.ContentAt(1, &hotel.City)
-		res.ContentAt(2, &hotel.State)
-		res.ContentAt(3, &hotel.Address)
-		res.ContentAt(4, &hotel.Name)
-		res.ContentAt(5, &hotel.Description)
+		err = res.Content(&hotel)
+		if err != nil {
+			app.logger.Printf("Error occurred parsing hotel: %s", err)
+			writeJsonFailure(w, 500, err)
+			return
+		}
 		respData.Data = append(respData.Data, hotel)
 	}
+	b, err := json.Marshal(qp)
+	if err != nil {
+		app.logger.Printf("Failed to marshal search query: %s", err)
+	}
+	if len(respData.Data) == 0 {
+		if len(b) > 0 {
+			respData.Context.Add(fmt.Sprintf("FTS search - scoped to: inventory.hotel (no results)\n%s", string(b)))
+		} else {
+			respData.Context.Add("FTS search - scoped to: inventory.hotel (no results)\n")
+		}
+		encodeRespOrFail(w, respData)
+		return
+	}
+	respData.Context.Add(fmt.Sprintf("FTS search - scoped to: inventory.hotel within fields %s\n%s",
+		strings.Join(hotelCols, ","), string(b)))
 
 	encodeRespOrFail(w, respData)
 }
 
-func main() {
-	var err error
+func envFlagString(envName, name, value, usage string) *string {
+	envValue := os.Getenv(envName)
+	if envValue != "" {
+		value = envValue
+	}
+	return flag.String(name, value, usage)
+}
 
-	// Connect to Couchbase
+func main() {
+	connStr := envFlagString("CB_HOST", "host", defaultCBHost,
+		"The connection string to use for connecting to the server")
+	username := envFlagString("CB_USER", "user", defaultCBUsername,
+		"The username to use for authentication against the server")
+	password := envFlagString("CB_PASS", "password", defaultCBPassword,
+		"The password to use for authentication against the server")
+	scheme := envFlagString("CB_SCHEME", "scheme", defaultCBScheme,
+		"The scheme to use for connecting to couchbase. Default to couchbases - set to couchbase:// when using couchbase community edition")
+	flag.Parse()
+
+	logrusLogger := logrus.New()
+	logrusLogger.SetFormatter(&logrus.JSONFormatter{})
+	logrusLogger.SetOutput(os.Stdout)
+	logrusLogger.SetLevel(logrus.InfoLevel)
+
+	// Uncomment to enable the Go SDK logging.
```

Comment:
Why not enable this anyway? Or would it make the logs too noisy when running the server?

Reply:
Yeah it adds a whole load of logging that I don't think users will actually be interested in unless they need to debug something.